tests/unit/completion/test_completionwidget.py

```diff
@@ -0,0 +1,152 @@
+    completionview.set_model(filtermodel)
+    assert completionview.model() == filtermodel
+    for i in range(model.rowCount()):
+        assert completionview.isExpanded(filtermodel.index(i, 0))
+
+
+def test_set_pattern(completionview):
+    model = sortfilter.CompletionFilterModel(base.BaseCompletionModel())
+    model.set_pattern = unittest.mock.Mock()
+    completionview.set_model(model)
+    completionview.set_pattern('foo')
+    model.set_pattern.assert_called_with('foo')
+
+
+def test_maybe_resize_completion(completionview, config_stub, qtbot):
+    """Ensure completion is resized only if shrink is True."""
+    with qtbot.assertNotEmitted(completionview.resize_completion):
+        completionview.maybe_resize_completion()
+    config_stub.data = {'completion': {'shrink': True}}
+    with qtbot.waitSignal(completionview.resize_completion):
+        completionview.maybe_resize_completion()
+
+
+@pytest.mark.parametrize('tree, count, expected', [
+    ([['Aa']], 1, 'Aa'),
+    ([['Aa']], -1, 'Aa'),
+    ([['Aa'], ['Ba']], 1, 'Aa'),
+    ([['Aa'], ['Ba']], -1, 'Ba'),
+    ([['Aa'], ['Ba']], 2, 'Ba'),
+    ([['Aa'], ['Ba']], -2, 'Aa'),
+    ([['Aa', 'Ab', 'Ac'], ['Ba', 'Bb'], ['Ca']], 3, 'Ac'),
+    ([['Aa', 'Ab', 'Ac'], ['Ba', 'Bb'], ['Ca']], 4, 'Ba'),
+    ([['Aa', 'Ab', 'Ac'], ['Ba', 'Bb'], ['Ca']], 6, 'Ca'),
+    ([['Aa', 'Ab', 'Ac'], ['Ba', 'Bb'], ['Ca']], 7, 'Aa'),
+    ([['Aa', 'Ab', 'Ac'], ['Ba', 'Bb'], ['Ca']], -1, 'Ca'),
+    ([['Aa', 'Ab', 'Ac'], ['Ba', 'Bb'], ['Ca']], -2, 'Bb'),
+    ([['Aa', 'Ab', 'Ac'], ['Ba', 'Bb'], ['Ca']], -4, 'Ac'),
+    ([[], ['Ba', 'Bb']], 1, 'Ba'),
+    ([[], ['Ba', 'Bb']], -1, 'Bb'),
+    ([[], [], ['Ca', 'Cb']], 1, 'Ca'),
+    ([[], [], ['Ca', 'Cb']], -1, 'Cb'),
+    ([['Aa'], []], 1, 'Aa'),
+    ([['Aa'], []], -1, 'Aa'),
+    ([['Aa'], [], []], 1, 'Aa'),
+    ([['Aa'], [], []], -1, 'Aa'),
+])
+def test_on_next_prev_item(tree, count, expected, completionview,
+                           config_stub, qtbot, monkeypatch,
+                           status_command_stub):
+    """Test that on_next_prev_item moves the selection properly.
+
+    Args:
+        tree: Each entry array represents a completion category, with each
+              string being an item under that category.
+        count: Number of times to go forward (or back if negative).
+        expected: item data that should be selected after going back/forward.
+    """
+    model = base.BaseCompletionModel()
+    for catdata in tree:
+        cat = QStandardItem()
+        model.appendRow(cat)
+        for name in catdata:
+            cat.appendRow(QStandardItem(name))
+    filtermodel = sortfilter.CompletionFilterModel(model)
+    completionview.set_model(filtermodel)
+    # actually calling show() will pop a window during the test, so just fool
+    # the completionview into thinking it is visible instead
+    monkeypatch.setattr(completionview, 'isVisible', lambda: True)
+    for i in range(abs(count)):
```

Comment:
Please do `for _ in ...` here (`_` is commonly used in Python to say "this is intended to be unused")